android/src/main/java/com/polidea/flutter_ble_lib/MultiCharacteristicsResponse.java

```diff
@@ -4,21 +4,34 @@
 import com.polidea.multiplatformbleadapter.Service;
 
 import java.util.List;
+import java.util.UUID;
 
 public class MultiCharacteristicsResponse {
     private final List<Characteristic> characteristics;
-    private Service service;
+    private int serviceId;
+    private UUID serviceUuid;
 
     public MultiCharacteristicsResponse(List<Characteristic> characteristics, Service service) {
         this.characteristics = characteristics;
-        this.service = service;
+        this.serviceId = service.getId();
+        this.serviceUuid = service.getUuid();
+    }
```

Comment:
How about:
```
    public MultiCharacteristicsResponse(List<Characteristic> characteristics, @Nullable Service service) {
        this.characteristics = characteristics;
        this.serviceId = service != null ? service.getId() : -1;
        this.serviceUuid = service != null ? service.getUuid() : null;
    }
```

Reply:
Service in this constructor should not be Nullable. The new constructor is because I'm retrieving service's UUID and ID from field in Characteristic, which does not have a reference to Service object.